### ISSUE TITLE
UP006 not catching anything

### ISSUE THREAD

**Issue by Garrett-R** [astral-sh/ruff#6689]
## Steps to repro

- create an empty directory
- create a Python 3.11 virtualenv in that directory and activate it (I have Python 3.11.4)
- Do `pip install ruff`  (got version `0.0.285`)
- Create this file called `repro.py`:

```python
from typing import List

def my_func(x: List[str]) -> None:
    pass
```

- Run the command: `ruff check --select ARG001,UP006 repro.py`

## Expected Behavior

There should be a [UP006 error](https://beta.ruff.rs/docs/rules/non-pep585-annotation/) among the output.

## Actual Behavior

The output is:
```python
$ ruff check --select ARG001,UP006 repro.py
repro.py:3:13: ARG001 Unused function argument: `x`
Found 1 error.
```

## System

- Ubuntu 22.04.1

**Comment by Garrett-R** [comment]
Oh, I see, I need to set [`target-version`](https://beta.ruff.rs/docs/settings/#target-version) to my current Python version.
